fpt-egui/src/main.rs

```diff
@@ -550,7 +543,23 @@ impl FPT {
 
     fn central_panel(&mut self, ctx: &Context, ui: &mut Ui) {
         if !self.gb.cpu().paused() {
-            self.emulator(ui);
+            let buttons = Buttons {
+                a: ctx.input(|i| i.key_down(Key::A)),
+                b: ctx.input(|i| i.key_down(Key::S)),
+                select: ctx.input(|i| i.key_down(Key::D)),
+                start: ctx.input(|i| i.key_down(Key::F)),
+                up: ctx.input(|i| i.key_down(Key::K)),
+                down: ctx.input(|i| i.key_down(Key::J)),
+                left: ctx.input(|i| i.key_down(Key::H)),
+                right: ctx.input(|i| i.key_down(Key::L)),
+            };
+            self.gb.bus_mut().set_buttons(buttons);
```

Comment:
is there a way we can avoid fpt-egui accessing the bus directly?

Reply:
well, either the client has to write *into* the gameboy, or the gameboy has to reach out and ask the client what the state of the buttons is. and it ultimately will need to touch the bus since it is a hardware register... I don't have strong opinions whether the client accesses the bus directly or indirectly. are you suggesting a method in e.g. the gameboy itself to set the buttons like `self.gb.set_buttons` vs `self.gb.bus_mut().set_buttons` or something else?